src/www/ui/ui-report-conf.php

```diff
@@ -179,9 +179,9 @@ function allReportConfiguration($uploadId, $groupId)
                      $obData['text'].'</textarea></td><td>';
       foreach ($obData['license'] as $value) {
         if (!empty($excludedObligations[$obTopic]) && in_array($value, $excludedObligations[$obTopic])) {
-          $tableRows .= '<input type="checkbox" name="obLicenses['.$obTopic.'][]" value="'.$value.'" checked> '.$value.'<br />';
+          $tableRows .= '<input type="checkbox" name="obLicenses['.str_replace(array('[',']'),array('&!;',';!&'),$obTopic).'][]" value="'.$value.'" checked> '.$value.'<br />';
```

Comment:
Instead of manually encoding characters, can we use `urlencode()` instead as it already handles such characters which cause issues and can easily be decoded with `urldecode()`?

Reply:
Agreed! made changes please have a look.